api/api.go

```diff
@@ -289,6 +293,12 @@ func (api *Server) GetServerMeta(ctx context.Context,
 // SendAction is the API to send an action to blockchain.
 func (api *Server) SendAction(ctx context.Context, in *iotexapi.SendActionRequest) (res *iotexapi.SendActionResponse, err error) {
 	log.L().Debug("receive send action request")
+	if len(in.Action.SenderPubKey) == secp256pubKeyLength-1 {
```

Comment:
Let's try not to tamper user data, but when verifying the signature (action.Verify), we do this massage on pk and sig before calling the verification cgo function. How do you think?

Reply:
https://github.com/iotexproject/go-pkgs/pull/9 already did this,but test failed,I think the action's hash is not same as original action,have to hard fork to make this work